### PR TITLE
fix sqs auth, tests, hard-coded region

### PR DESCRIPTION
**Pull request by yaron2**
This PR does the following:

* Fixes AWS auth to receive credentials from user container and not depend on env vars / credentials file passed to the KEDA controller
* Fixes hardcoded region
* Refactors ```GetAwsSqsQueueLength``` into scaler
* Fixes SQS test
* Minor code formatting changes
* Adds AWS SQS support to Readme

### CODE REVIEW

File: pkg/scalers/aws_sqs_queue_scaler.go [hunk 1]
```diff
@@ -52,25 +61,38 @@ func parseAwsSqsQueueMetadata(metadata, resolvedEnv map[string]string) (*awsSqsQ
...
+	} else {
+		return nil, fmt.Errorf("no region given")
+	}
+
+	if val, ok := resolvedEnv["awsAccessKeyID"]; ok && val != "" {
+		meta.awsAccessKeyID = val
+	} else {
+		meta.awsAccessKeyID = awsAccessKeyIDEnvVar
+	}
+
```
Comment (ahmelsayed) [inline pkg/scalers/aws_sqs_queue_scaler.go]: `meta.awsAccessKeyID` and `meta.awsSecretAccessKey` should have the actual value of the keyId and secret, right? 

I think this code (as well as the one for `awsSecretAccessKey` below) should be something like

```suggestion
    var keyName string
	if keyName, ok := metadata["awsAccessKeyID"]; !ok || keyName == "" {
		keyName = awsAccessKeyIDEnvVar
	}
	
	if val, ok := resolvedEnv[keyName]; ok && val != "" {
		meta.awsAccessKeyID = val
	} else {
		return nil, fmt.Errorf("cannot find awsAccessKeyId named %s in pod environment", keyName)
	}
```

right?

Comment (yaron2) [reply pkg/scalers/aws_sqs_queue_scaler.go]: Correct

File: pkg/scalers/aws_sqs_queue_scaler.go [hunk 2]
```diff
@@ -52,25 +61,47 @@ func parseAwsSqsQueueMetadata(metadata, resolvedEnv map[string]string) (*awsSqsQ
...
+	} else {
+		return nil, fmt.Errorf("no region given")
+	}
+
+	if val, ok := resolvedEnv["awsAccessKeyID"]; ok && val != "" {
+		meta.awsAccessKeyID = val
+	} else {
+		meta.awsAccessKeyID = awsAccessKeyIDEnvVar
+	}
+    var keyName string
```
Comment (ahmelsayed) [inline pkg/scalers/aws_sqs_queue_scaler.go]: should use tab instead of 4 spaces.

File: pkg/scalers/aws_sqs_queue_scaler.go [hunk 3]
```diff
@@ -52,25 +61,47 @@ func parseAwsSqsQueueMetadata(metadata, resolvedEnv map[string]string) (*awsSqsQ
...
+	
+	if val, ok := resolvedEnv[keyName]; ok && val != "" {
+		meta.awsAccessKeyID = val
+	} else {
+		return nil, fmt.Errorf("cannot find awsAccessKeyId named %s in pod environment", keyName)
+	}
+	if val, ok := resolvedEnv["awsSecretAccessKey"]; ok && val != "" {
+		meta.awsSecretAccessKey = val
+	} else {
+		meta.awsSecretAccessKey = awsSecretAccessKeyEnvVar
```
Comment (ahmelsayed) [inline pkg/scalers/aws_sqs_queue_scaler.go]: same as with the `awsAccessKeyId` above, get the value from resolvedEnv based on the env var name.

File: pkg/scalers/aws_sqs_queue_scaler.go (hunk 2 above)
Comment (yaron2) [reply pkg/scalers/aws_sqs_queue_scaler.go]: That's Github's code suggestion fault :)

File: pkg/scalers/aws_sqs_queue_scaler.go (hunk 3 above)
Comment (yaron2) [reply pkg/scalers/aws_sqs_queue_scaler.go]: done

**Pull request merged**